### ISSUE TITLE
Server Error when updating argument

### ISSUE THREAD

**Issue by SerhiiCho** [brendt/rfc-vote#191]
Steps to reproduce:

1. Click the ellipsis button on your argument
2. Click "Edit" from a dropdown menu (popup)
3. Click a green button "Save"

I get a server error in a JavaScript popup window:

![Screenshot 2023-08-26 at 19 20 03](https://github.com/brendt/rfc-vote/assets/35465417/b7e638e7-5164-474c-a983-aeeaf2b57880)

**Comment by vsergiu93** [comment]
@SerhiiCho, so this happens only on https://rfc.stitcher.io/? I tried to reproduce it locally but I wasn't able to do so.

**Comment by SerhiiCho** [comment]
> so this happens only on https://rfc.stitcher.io/?

Yes @vsergiu93, only on stitcher.io. @brendt might be able to just look in the log file when he's available and fix it

**Comment by brendt** [comment]
I'll have to look into this on Monday, thanks for the detailed steps!

**Comment by Erol314** [comment]
So, I did few tests=)

- This only happens to default methods rfc, other 2 Rfcs work fine 
- Not only argument edit returns 500, but upvoting/downvoting too

**Comment by brendt** [comment]
Ok, so this is the error 😅

```
[2023-08-29 11:21:54] production.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 1343488 bytes) {"userId":1,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 1343488 bytes) at /home/forge/rfc.stitcher.io/releases/20230829-111518/vendor/laravel/framework/src/Illuminate/Database/Connection.php:423)
[stacktrace]
#0 {main}
"} 
```

Not much info :/ Although I think I can guess what's going on: we're loading all arguments of that RFC in memory, together with a bunch of other relations. It only happens on this RFC, because this one has much more arguments. It also explains why local works fine (no memory limits).

I'll work on a fix

**Comment by brendt** [comment]
Actually, I would like some input on the best approach before diving into code:

We have 30MB of memory on this page:

<img width="2048" alt="Screenshot 2023-08-29 at 13 25 30" src="https://github.com/brendt/rfc-vote/assets/6905297/71814ecb-42f6-4e11-96f8-1b22ce01cd66">

All subsequent livewire requests take 10-11MB (per request)

<img width="2048" alt="Screenshot 2023-08-29 at 13 25 51" src="https://github.com/brendt/rfc-vote/assets/6905297/54dc19af-b20d-4c5d-bd0b-6a53e472cf48">

As you can see, we only have 20 queries, that's because everything is properly eager-loaded so we don't run into performance issues. However, because of that, we run into memory issues 😅

A couple of possible fixes:

- Increase the memory limit on the server: the easiest fix, but it's more like applying a bandaid, I'd like to make the page more performant overall anyway
- Add pagination for arguments in the Livewire component?
- Extract each argument to a separate Livewire component so that doesn't have to refresh the whole list (this probably won't work, I remember trying this a while ago, it's a mess)
- Try to optimise model usage so that we don't have to load deeply nested relations
- Other ideas?

I think the best approach for now is pagination, but would love some input

**Comment by SerhiiCho** [comment]
Pagination is necessary here, but we can also optimize model usage

**Comment by brendt** [comment]
It's fixed thanks to pagination.
